### PR TITLE
Adds colors back into tests.

### PR DESCRIPTION
**Pull request by supersimple**
I tried to use existing functions like IO.ANSI.format/2 to minimize changed code. The second arg for that function is whether or not to keep the ansi formatting. 

I am using a function that I put in Hex.Shell to make that false in test.

I am not sure this is the cleanest implementation, so feedback is welcomed.

screen shot 2018-04-13 at 5 06 45 pm

### CODE REVIEW

File: test/support/case.ex [hunk 1]
```diff
@@ -278,9 +278,9 @@ defmodule HexTest.Case do
       Mix.ProjectStack.clear_stack()
     end
 
-    if context[:ansi_enabled] do
-      Application.put_env(:elixir, :ansi_enabled, true)
-      on_exit(fn -> Application.put_env(:elixir, :ansi_enabled, false) end)
```
Comment (ericmj) [inline test/support/case.ex]: Do we still need this?

File: test/mix/tasks/hex.outdated_test.exs [hunk 2]
```diff
@@ -78,7 +78,7 @@ defmodule Mix.Tasks.Hex.OutdatedTest do
     end
   end
 
-  @tag ansi_enabled: true
```
Comment (ericmj) [inline test/mix/tasks/hex.outdated_test.exs]: This should rather change the Hex ansi setting, so we may need to move the setting to `Hex.Server` instead checking based on `Mix.env` at compile time.

File: test/hex/resolver/backtracks_test.exs [hunk 3]
```diff
@@ -50,7 +50,7 @@ defmodule Hex.Resolver.BacktracksTest do
 
   defp format(message) do
     message
-    |> IO.ANSI.format()
+    |> IO.ANSI.format(Hex.Shell.ansi_enabled?())
     |> IO.iodata_to_binary()
```
Comment (ericmj) [inline test/hex/resolver/backtracks_test.exs]: Let's call `Hex.Shell.format` instead where `ansi_enabled?()` is a default argument.

File: test/mix/tasks/hex.outdated_test.exs (hunk 2 above)
Comment (supersimple) [reply test/mix/tasks/hex.outdated_test.exs]: it turns out this was no longer being used.

**Comment by ericmj** [comment]
Looks good. Let's change the remaining places where we use IO.ANSI.format + ansi_enabled instead of Hex.Shell.format and then we should be good to go.

**Comment by ericmj** [comment]
Thank you!

**Pull request merged**